### ISSUE TITLE
Move to Typescript and Styled components

### ISSUE THREAD

**Issue by Algram** [Algram/ytdl-webserver#30]
I'd like to move this project to Typescript and Styled components. Feel free to contribute! The linting should be done with eslint.

**Comment by murbar** [comment]
I can help with styled components.

**Comment by emanuelmd** [comment]
I can port it to TypeScript

**Comment by emanuelmd** [comment]
The frontend is fairly doable since it runs through webpack and we can easily add a ts-loader in there.

The server will be a bit trickier, I'm afraid you have to move it out of `src/` or somehow compile it differently because node won't be able to run `.ts` files.

You have two options here
- Use `ts-node` that just runs `.ts` files. Although I'm not a big fan of this, last time I tried it, it was buggy
- Separate the server to it's own folder along with it's own `package.json` and `tsconfig.json`

**Comment by emanuelmd** [comment]
Also, it'd be a great idea to move everything to `async/await` since you're doing TypeScript. No need to resort to callbacks.

**Comment by Algram** [comment]
Yeah I figured this would be a problem. We could also do server side rendering and then compile the project all at once with a single `tsconfig.json`.

For the refactoring: I wanted to do that for a long time now but haven't found the time for it yet, feel free to do this. As long as you adhere to proper linting and keep the core functionality I will approve of it.

**Comment by wilsonj806** [comment]
Just as a quick heads up, Babel handles [transpiling TypeScript](https://babeljs.io/docs/en/babel-preset-typescript) now as well, although I haven't handled transpiling both `.ts` and `.tsx` before but it should be fine.

**Comment by Algram** [comment]
Yep this works, I am doing this at work already in a prod environment.
